app/addons/news/components.js

```diff
@@ -9,17 +9,81 @@
 // WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 // License for the specific language governing permissions and limitations under
 // the License.
+import app from "../../app";
 
 import React from "react";
 
-const NewsPage = () => {
+const LoadNewsButton = ({ showNews, isChecked, toggleCookieSave }) => {
   return (
-    <div id="news-page" className="">
-      <iframe src="https://blog.couchdb.org" width="100%" height="100%"></iframe>
+    <div>
+      <p>
+        When you click this button, you are requesting content and sharing your IP address with <a href="https://blog.couchdb.org/">blog.couchdb.org</a> which is edited by the Apache CouchDB PMC and maintained by <a href="https://wordpress.com/">wordpress.com</a>.
+      </p>
+      <p>
+        If you don’t want to share your IP address, do not click the button.
+      </p>
+      <button className="btn btn-primary" onClick={showNews}>Load News</button>
+      <label className="news-checkbox">
+        <input type="checkbox"
+          checked={isChecked}
+          onChange={toggleCookieSave}
+        />
+        Remember my choice
+      </label>
     </div>
   );
 };
 
+class NewsPage extends React.Component {
+  constructor (props) {
+    super(props);
+    this.showNews = this.showNews.bind(this);
+    this.toggleCookieSave = this.toggleCookieSave.bind(this);
+
+    const hasCookie = !!app.utils.localStorageGet('allow-IP-sharing');
+
+    this.state = {
+      showNews: hasCookie ? true : false,
+      hasCookie
+    };
+  }
+
+  showNews() {
+    this.setState({ showNews: true });
+  }
+
+  toggleCookieSave() {
+    if (!this.state.hasCookie) {
+      this.setState(() => {
+        return { hasCookie: true };
+      });
+      app.utils.localStorageSet('allow-IP-sharing', true);
+
+    } else {
+      this.setState(() => {
+        return { hasCookie: false };
+      });
+      app.utils.localStorageSet('allow-IP-sharing', false);
+    }
+  }
+
+  render() {
+    return (
+      <div id="news-page" className="">
+        {this.state.showNews ?
```

Comment:
I don't feel very comfortable with `let`ting and reassigning, are you sure about that?

(sorry for the unsoliticited feedback)

Reply:
@dottorblaster I'm also not really comfortable with that, because the button is not 'news'. I changed it back and save the iframe in the news variable. What do you think?

Reply:
@hulkoba much better now, thanks for taking the comment under consideration :)

Reply:
Sorry, I'm not a big fan of that implementation. If you look in all the other Fauxton code we very really have the conditional in the jsx. I feel that is a lot more error-prone and much harder to read. If you don't link calling it news you can call it something else.